subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -1933,6 +2012,7 @@ static int handle_request(struct zoap_packet *request,
 	struct lwm2m_engine_context context;
 	int observe = -1; /* default to -1, 0 = ENABLE, 1 = DISABLE */
 	bool discover = false;
+	struct block_context *ctx = NULL;
```

Comment:
There is already context variable, and now you introduced ctx. These look very similar to my eyes, could you rename this one to something like block_ctx or similar in order to avoid confusion.